cmd/nitro/init.go

```diff
@@ -795,6 +800,11 @@ func openInitializeChainDb(ctx context.Context, stack *node.Node, config *NodeCo
 		return chainDb, l2BlockChain, err
 	}
 
+	if config.Init.RebuildLocalWasm != "false" {
+		// In order to rebuild wasm store correctly we have to use force option
+		return rebuildLocalWasm(ctx, config, l2BlockChain, chainConfig, chainDb, wasmDb, "force")
```

Comment:
I think this should just use the Init.RebuildLocalWasm flag.
If we're downloading a database without WASM - when creating the new database there won't be a wasm rebuild and so we'll rebuild it here.
If we're downloading a database with wasm (which we had to specifically enable) the whole point is to avoid local rebuilding wasm phase.

Reply:
in the first scenario you mentioned, consider this variation- if we had previously attempted rebuilding and the node was stopped we would have a non-empty and non-done rebuilding position and after we download a new database without wasm chaindb gets updated and now when rebuilding starts with old wasm but new chaindb we might miss some programs right? Hence to handle this I was always setting it to force.

But I agree that we could just let the user decide this

Reply:
People are only expected to download a new database when they sync a new node.
For weird/unexpected cases - we always have fallback to compiling wasms on-demand